app/assets/purs/src/EntryPoints/Comments.purs

```diff
@@ -0,0 +1,106 @@
+module EntryPoints.Comments
+  ( boot
+  ) where
+
+import Prelude
+
+import Data.Array as Array
+import Data.Maybe (Maybe(..))
+import Elmish (BootRecord, ComponentDef, Dispatch, ReactElement, Transition, handleMaybe)
+import Elmish.Foreign (Foreign, readForeign)
+import Elmish.HTML.Styled as H
+import Foreign.Object as Obj
+import Utils.Boot (bootOrPanic)
+-- import Utils.API as API
+
+boot :: BootRecord Foreign
+boot = bootOrPanic
+  { def: Just <<< def
+  , diagnosticName: "Demo"
+  }
+
+data Message
+  = Create
+  | CommentTextChanged String
+  | CommenterChanged String
+
+type Props =
+  { comments :: Array Comment
+  }
+
+type Comment = 
+  { commenter :: String 
+  , comment :: String 
+  }
+
+type State = 
+  { commenter :: String
+  , newComment :: String
+  , comments :: Array Comment
+  }
+
+def :: Props -> ComponentDef Message State
+def props =
+  { init, update, view }
+  where
+    init :: Transition Message State
+    init = 
+      pure
+        { newComment: ""
+        , commenter : ""
+        , comments: props.comments }
+
+
+    update :: State -> Message -> Transition Message State
+    update state = case _ of
+      CommentTextChanged updatedComment ->
+        pure state 
+          { newComment = updatedComment }
+
+      CommenterChanged updatedCommenter ->
+        pure state 
+          { commenter = updatedCommenter }
+        
+      Create ->
+        pure state 
+          {
+            comments = Array.snoc state.comments { commenter: state.commenter, comment: state.newComment }
+          , commenter = ""
+          , newComment = ""
+          }
+
+    view :: State -> Dispatch Message -> ReactElement
+    view state dispatch = H.fragment
+                              [ H.h3 "" "Comments"
+                              , H.ul "" $
+                                (\comment -> H.li ""
+                                              [ H.div "row" 
+                                                [ H.text comment.comment ]
+                                              , H.div "row"
+                                                [ H.text comment.commenter ] 
+                                              ]
+                                ) <$> state.comments
```

Comment:
I believe this could be rewritten using anonymous arguments (untested)
```suggestion
                                 H.li ""
                                [ H.div "row" 
                                  [ H.text _.comment ]
                                , H.div "row"
                                  [ H.text _.commenter ] 
                                ]
                                 <$> state.comments
```

Reply:
I tried it and got a type error:
```
Could not match type

    { comment :: t1
    | t2
    }
    -> t1

  with type

    String


while checking that type t0 -> t1
  is at least as general as type String
while checking that expression \$4 ->
                                 $4.comment
  has type String
in value declaration def

where t0 is an unknown type
      t1 is an unknown type
      t2 is an unknown type 
```